mobile/library/common/internal_engine.h

```diff
@@ -134,7 +133,7 @@ class InternalEngine : public Logger::Loggable<Logger::Id::main> {
   Event::Dispatcher* event_dispatcher_{};
   Stats::ScopeSharedPtr client_scope_;
   Stats::StatNameSetPtr stat_name_set_;
-  envoy_engine_callbacks callbacks_;
+  std::unique_ptr<InternalEngineCallbacks> callbacks_;
```

Comment:
why does it need to be a unique_ptr instead of just `InternalEngineCallbacks`?

Reply:
`EngineBuilder` uses `std::unique_ptr<InternalEngineCallbacks>` and when we call `EngineBuilder::build()`, the callback owner will be moved to `InternalEngine`. If we made it an `InternalEngineCallbacks`, we would need to make a copy instead of moving the pointer.

Reply:
got it, thanks for the explanation